api/src/db/models/opportunity_models.py

```diff
@@ -81,6 +81,14 @@ def applicant_types(self) -> list[ApplicantType]:
         # Helper method for serialization of the API response
         return [a.applicant_type for a in self.link_applicant_types]
 
+    def __repr__(self) -> str:
+        return (
+            f"<Opportunity(opportunity_id={self.opportunity_id!r}, "
+            f"opportunity_number={self.opportunity_number!r}, "
+            f"opportunity_title={self.opportunity_title!r}, "
+            f"agency={self.agency!r}, category={self.category!r})"
+        )
```

Comment:
There might be an easier to maintain way that we could use - I'm going to break a lot of this with an upcoming set of refactors to the models. The models already have a `for_json` method that converts a model to a dictionary. If we instead add a repr function like this to the base class, we wouldn't need to add this to every class:

```
def __repr__(self) -> str:

     values = []
     for k, v in self.for_json().items():
         # Note that I see you added a format method to a lot of classes, probably call that on v here
         values.append(f"{k}={v!r}")

      
      return f"<{self.__class__.__name__}({','.join(values)})"
```

Reply:
yea the above worked - thanks!

Reply:
Added a couple factories!